lib/rouge/lexers/kotlin.rb

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Kotlin < RegexLexer
+      title "Kotlin"
+      desc "Kotlin <http://kotlinlang.org>"
+
+      tag 'kotlin'
+      filenames '*.kt'
+      mimetypes 'text/x-kotlin'
+
+      keywords = %w(
+        abstract annotation as break by catch class companion const 
+        constructor continue crossinline do dynamic else enum
+        external false final finally for fun get if import in infix
+        inline inner interface internal is lateinit noinline null
+        object open operator out override package private protected
+        public reified return sealed set super tailrec this throw
+        true try typealias typeof val var vararg when where while
+        yield
+      )
+
+      name = '@?[_\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Nl}]' +
+             '[\p{Lu}\p{Ll}\p{Lt}\p{Lm}\p{Nl}\p{Nd}\p{Pc}\p{Cf}\p{Mn}\p{Mc}]*'
+
+      id = "(#{name}|`#{name}`)"
```

Comment:
``` irb
=> /foo/i
irb(main):004:0> outer = /bar#{inner}baz/
=> /bar(?i-mx:foo)baz/
irb(main):005:0> 
```

More or less it automatically surrounds with non-capturing parens and preserves flags. Backslashing is also a little more straightforward.